lib/auth/webauthntypes/webauthn.go

```diff
@@ -376,3 +379,46 @@ type Credential protocol.Credential
 type AuthenticationExtensionsClientOutputs struct {
 	AppID bool `json:"appid,omitempty"`
 }
+
+// SessionData is a clone of [webauthn.SessionData], materialized here to keep a
+// stable JSON marshal/unmarshal representation and add extensions.
+type SessionData struct {
+	// Raw challenge used for the ceremony.
+	Challenge []byte `json:"challenge,omitempty"`
+	// Raw User ID.
+	UserId []byte `json:"userId,omitempty"`
+	// Raw Credential IDs of the credentials allowed for the ceremony.
+	AllowCredentials [][]byte `json:"allowCredentials,omitempty"`
+	// True if resident keys were required by the server / Relying Party.
+	ResidentKey bool `json:"residentKey,omitempty"`
+	// Requested user verification requirement, either "discouraged" or
+	// "required".
+	// An empty value is treated equivalently to "discouraged".
+	UserVerification string `json:"userVerification,omitempty"`
+	// ChallengeExtensions are Teleport extensions that apply to this webauthn session.
+	ChallengeExtensions mfav1.ChallengeExtensions `json:"challenge_extensions,omitempty"`
```

Comment:
Maybe, so we can differentiate set/unset:

```suggestion
	ChallengeExtensions *mfav1.ChallengeExtensions `json:"challenge_extensions,omitempty"`
```

It doesn't look like this is needed now, but the pointer is the more "usual" design.